### PR TITLE
[Tests] Dump duration of Pytests.

### PR DESCRIPTION
**Pull request by anijain2305**
Dumping test durations can help identify any misbehaving tests.

@tqchen @icemelon9 @wweic @broune

### CODE REVIEW

**Comment by u99127** [comment]
We need to measure this I agree. 

There's a simpler way of achieving this this - I would suggest using PYTEST_ADDOPTS instead in docker/build.sh and docker/bash.sh . That then means we would be able to globally change this instead of changing this in every single script.

I've been playing with that for some other items that I've been playing with in pytest - pull requests will follow.

**Comment by anijain2305** [comment]
@u99127 I can close this PR then. I think your PR will supersede this.

**Comment by broune** [comment]
tldr: how about having a run_pytets.sh file that takes care of how to run pytest?

1) I agree with Ramana that it is unfortunate to duplicate this into every invocation of pytest.
2) It is preferable to have it be very clear how to reproduce a command line run in CI and Docker in some other context and dependencies on environment variables make that harder.
3)  Thinking ahead, we also want to run e.g. pytest-parallel, as the people on this thread have discussed before, which will also involve a parameter to pytest to trigger it to run N-ways parallel (providing the N) and ideally that will be done the same way. In this case we probably don't want to rebuild a different docker for each level of parallelism, which may vary by machine and other factors, so it's not ideal to handle pytest parameters via the Docker build from that perspective.

I think we can satisfy 1-3 at the same time by having a run_pytest.sh script that is used to run pytest every time and can print out the actual command used. It can be changed without rebuilding the Dockers, it's in one place, so no duplication, it's easy to tell what it's doing and it gets us out of the current business (also before this changer) of encoding into each file how pytest is run. We can even have it print out a command line for how it was run, including any relevant environment variables like TVM_FFI and indeed PYTEST_ADDOPTS if they are non-empty.

What do you think?

**Comment by tqchen** [comment]
opting for simplicity, perhaps we can just export the necessary env variables in the test script if it is not defined. Then allow passing in PYTEST_ADDOPTS env variable in the bash script as @u99127 suggested

**Comment by broune** [comment]
Re @tqchen, which file does "the test script" refer to?

**Comment by tqchen** [comment]
e.g. add exports here https://github.com/apache/incubator-tvm/blob/master/tests/scripts/task_python_integration.sh#L22

**Comment by broune** [comment]
What I like about having a single script that knows how to run pytest on TVM is that it prevents duplication of the logic for this, including environment variables, into the 8 files changed in this change here and then the docker and makes it easy to change in one place without recreating dockers. This logic is currently out of sync across these files and it is not clear how much of that is on purpose and how much is drift (the motivation for the settings is not documented in the files). Though thinking more about it I'm mainly happy that we get to see test times and whichever way that achieves that is an improvement in my view.

**Comment by anijain2305** [comment]
Closing this one as @u99127 plans to send another PR for this. If not, please let me, I will send one with PYTEST_ADDOPTS.
